src/main/java/ru/test/restservice/api/FileController.java

```diff
@@ -0,0 +1,103 @@
+
+    // Функция объединения списков (надо бы вынести в класс utils)
+    public <T> ArrayList<T> joinLists(
+            final List<T> listA,
+            final List<T> listB) {
+        boolean aEmpty = (listA == null) || listA.isEmpty();
+        boolean bEmpty = (listB == null) || listB.isEmpty();
+        if (aEmpty & bEmpty) {
+            return new ArrayList<>();
+        } else if (aEmpty) {
+            return new ArrayList<>(listB);
+        } else if (bEmpty) {
+            return new ArrayList<>(listA);
+        } else {
+            ArrayList<T> result = new ArrayList<>(
+                    listA.size() + listB.size());
+            result.addAll(listA);
+            result.addAll(listB);
+            return result;
+        }
+    }
+
+    // Ловлю передаваемые файлы, создаю их в файловой системе и возвращаю соответствующий http код
+    @PostMapping(value = "/upload", headers = "content-type=multipart/*")
+    public ResponseEntity<?> handleFileUpload(@RequestParam("file") MultipartFile file) {
+        try {
+            // Проверяю и на фронте, и на бэке разрешён ли файл (на фронте пользователь может удалить метод)
+            boolean isAllowedExtension = false;
+            for (String s : mask) {
+                isAllowedExtension = isAllowedExtension || Objects.requireNonNull(file.getOriginalFilename()).toLowerCase().endsWith(s);
+            }
+            if (isAllowedExtension) {
+                Path path = Paths.get("test/" + file.getOriginalFilename());
+                if (!Files.exists(path))
+                    Files.createFile(path);
+                Files.write(path, file.getBytes());
+            } else
+                return new ResponseEntity<>(HttpStatus.BAD_GATEWAY);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_GATEWAY);
+        }
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    // Обхожу все файлы в папке пользователя и возвращаю имя, тип и содержимое (для текстовых) через объект
+    @GetMapping("/filelist")
+    public FileList returnFileList() {
+        String[] files;
+        ArrayList<String> fileTypes = new ArrayList<>();
+        ArrayList<List<String>> contents = new ArrayList<>();
+        try {
+            File folder = new File("test");
+            // Выбираю только те, с которыми пользователь будет работать
+            files = folder.list((folder1, name) -> {
+                for (String s : mask)
+                    if (name.toLowerCase().endsWith(s))
+                        return true;
+                return false;
+            });
+            // Определяю тип и, если формат текстовый, читаю содержимое, иначе отдаю путь к изображению
+            for (int i = 0; i < (files != null ? files.length : 0); i++) {
+                boolean isTextFile = false;
+                for (String s : maskTextFile)
+                    isTextFile = isTextFile || files[i].endsWith(s);
```

Comment:
аналогично 
```suggestion
                boolean isTextFile = maskTextFile.stream().anyMatch(m -> files.[i].endsWith(m));
```
но вообще этот код дублируется, так что имеет смысл вынести в Utils